compiler.ts

```diff
@@ -312,145 +321,302 @@ function codeGenStmt(stmt: Stmt<[Type, Location]>, env: GlobalEnv): Array<string
+            },
+          };
+          var niass: Stmt<[Type, Location]> = {
+            a: [NONE, stmt.a[1]],
+            tag: "assignment",
+            destruct: makeId([NUM, stmt.a[1]], "idx" + stmt.id),
+            value: nid,
+          };
+          var Code_idstep = codeGenStmt(niass, env);
+
+          var list_lookup: Expr<[Type, Location]> = {
+            a: [NUM, stmt.a[1]],
+            tag: "bracket-lookup",
+            obj: stmt.iterable,
+            key: {
+              a: [NUM, stmt.a[1]],
+              tag: "id",
+              name: "idx" + stmt.id,
+            },
+          };
+
+          var tarname = "";
+          if (stmt.name.targets[0].target.tag === "id") {
+            tarname = stmt.name.targets[0].target.name;
+          }
+          // name = cur
+          var ass: Stmt<[Type, Location]> = {
+            a: [NONE, stmt.a[1]],
+            tag: "assignment",
+            destruct: makeId([NUM, stmt.a[1]], tarname),
+            value: list_lookup,
+          };
+          var Code_ass = codeGenStmt(ass, env);
+
+          // stop condition cur<step
+          var Expr_cond: Expr<[Type, Location]> = {
+            a: [BOOL, stmt.a[1]],
+            tag: "binop",
+            op: BinOp.Gte,
+            left: {
+              a: [NUM, stmt.a[1]],
+              tag: "id",
+              name: "idx" + stmt.id,
+            },
+            right: {
+              a: [NUM, stmt.a[1]],
+              tag: "literal",
+              value: {
+                tag: "num",
+                value: BigInt(stmt.iterable.contents.length),
+              },
+            },
+          };
+          var Code_cond = codeGenExpr(Expr_cond, env);
+
+          // if (stmt.index) {
+          //   var idass: Stmt<[Type, Location]> = {
+          //     a: [NONE, stmt.a[1]],
+          //     tag: "assignment",
+          //     destruct: makeId([NUM, stmt.a[1]], stmt.index),
+          //     value: { a: [NUM, stmt.a[1]], tag: "id", name: "idx" + stmt.id },
+          //   };
+          //   var Code_idass = codeGenStmt(idass, env);
+          //   return [
+          //     `
+          //     ${Code_iass.join("\n")}
+          //
+          //     (block
+          //       (loop
+          //         ${Code_idstep.join("\n")}
+          //         (br_if 1 ${Code_cond.join("\n")} ${decodeLiteral.join("\n")})
+          //
+          //         ${Code_ass.join("\n")}
+          //         ${Code_idass.join("\n")}
+          //         ${bodyStmts.join("\n")}
+          //         (br 0)
+          //     ))`,
+          //   ];
+          // }
+
+          return [
+            `
+            ${Code_iass.join("\n")}
+
+            (block
+              (loop
+                ${Code_idstep.join("\n")}
+                (br_if 1 ${Code_cond.join("\n")} ${decodeLiteral.join("\n")})
+
+                ${Code_ass.join("\n")}
+                ${bodyStmts.join("\n")}
+                (br 0)
+            ))`,
+          ];
```

Comment:
Missed a temp. guard and join removal here

Reply:
Fixed. Does it look right now?

Reply:
Looks good! Thanks